src/components/firstProfile/NameProfile.tsx

```diff
@@ -0,0 +1,103 @@
+import Logo from '../logo/Logo'
+import {
+  Box,
+  Button,
+  FormControl,
+  FormHelperText,
+  OutlinedInput,
+  Typography,
+} from '@mui/material'
+import { useState } from 'react'
+import useProfileData from '../../hooks/useProfileData'
+import axios from 'axios'
+// import { v4 as uuidv4 } from 'uuid'
+import { firstProfileStyles } from 'styles/firstProfileStyles'
+
+const FULLNAME_REGEX = /^[a-zA-Zа-яА-ЯёЁ\s\p{L}]{2,15}$/u
+
+const NameProfile = () => {
+  const { classes } = firstProfileStyles()
+  const [fullName, setFullName] = useProfileData<string>(
+    'profileData.fullName',
+    ''
+  )
+  const [error, setError] = useState<string>('')
+  const [isValid, setIsValid] = useState<boolean>(false)
+
+  // const userId = //
+  // const requestBody = {
+  //   id: userId,
+  //   name: fullName,
+  // }
+
+  const handleSubmit = async (e: { preventDefault: () => void }) => {
+    e.preventDefault()
+
+    if (fullName.trim() === '') {
+      setError('Please enter your full name')
+    } else if (!FULLNAME_REGEX.test(fullName)) {
+      setIsValid(true)
+    } else {
+      try {
+        await axios.post('api/addName', fullName)
```

Comment:
I think we save the data at the end when all the fields in the profile are filled in. I propose to clarify this point